React/Base/RCTConvert.m

```diff
@@ -525,6 +726,56 @@ + (UIColor *)UIColor:(id)json
     CGFloat g = ((argb >> 8) & 0xFF) / 255.0;
     CGFloat b = (argb & 0xFF) / 255.0;
     return [UIColor colorWithRed:r green:g blue:b alpha:a];
+  } else if ([json isKindOfClass:[NSDictionary class]]) {
+    NSDictionary *dictionary = json;
+    id value = nil;
+    if ((value = [dictionary objectForKey:@"semantic"])) {
+      if ([value isKindOfClass:[NSString class]]) {
+        NSString *semanticName = value;
+        UIColor *color = RCTColorFromSemanticColorName(semanticName);
+        if (color == nil) {
+          RCTLogConvertError(json, [@"a UIColor.  Expected one of the following values: " stringByAppendingString:RCTSemanticColorNames()]);
+        }
+        return color;
+      } else if ([value isKindOfClass:[NSArray class]]) {
+        for (id name in value) {
+          UIColor *color = RCTColorFromSemanticColorName(name);
+          if (color != nil) {
+            return color;
+          }
+        }
+        RCTLogConvertError(json, [@"a UIColor.  None of the names in the array were one of the following values: " stringByAppendingString:RCTSemanticColorNames()]);
+        return nil;
+      }
+      RCTLogConvertError(json, @"a UIColor.  Expected either a single name or an array of names but got something else.");
+      return nil;
+    } else if ((value = [dictionary objectForKey:@"dynamic"])) {
+      NSDictionary *appearances = value;
+      id light = [appearances objectForKey:@"light"];
+      UIColor *lightColor = [RCTConvert UIColor:light];
+      id dark = [appearances objectForKey:@"dark"];
+      UIColor *darkColor = [RCTConvert UIColor:dark];
+      if (lightColor != nil && darkColor != nil) {
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+        if (@available(iOS 13.0, *)) {
+          UIColor *color = [UIColor colorWithDynamicProvider:^UIColor * _Nonnull(UITraitCollection * _Nonnull collection) {
+            return collection.userInterfaceStyle == UIUserInterfaceStyleLight ? lightColor : darkColor;
```

Comment:
hmm would that mean that `UIUserInterfaceIdiomUnspecified` will return the dark color? can we flip this?

Reply:
Oh, good catch